Content.Server/Xenobiology/SpecimenDietComponent.cs

```diff
@@ -0,0 +1,74 @@
+using Content.Server.CombatMode;
+using Content.Server.Interaction;
+using Content.Server.Power.Components;
+using Content.Shared.ActionBlocker;
+using Content.Shared.Acts;
+using Content.Shared.DragDrop;
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Helpers;
+using Content.Shared.Verbs;
+using Content.Shared.Xenobiology;
+using Robust.Server.GameObjects;
+using Robust.Shared.Containers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Random;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.Utility;
+using Robust.Shared.ViewVariables;
+using System.Collections.Generic;
+using System.Numerics;
+using YamlDotNet.Core.Tokens;
+
+
+namespace Content.Server.Xenobiology
+{
+    [RegisterComponent]
+    public class SpecimenDietComponent : Component
+    {
+        [Dependency]
+        private readonly IRobustRandom _random = default!;
+        public override string Name => "SpecimenDietComponent";
+
+        //The diet is defined by a tag fed into it
+        [ViewVariables] [DataField("Diet")] public string[] DietPick { get; set; } = default!;
+
+        public string SelectedDiet = default!;
+
+        public int GrowthState = 0; //0 for embryo, 5 for mature specimen. Anything in-between is embryo state.
+        protected override void Initialize()
+        {
+            IoCManager.Resolve<IRobustRandom>();
+            base.Initialize();
+            SelectDiet();
+        }
+
+        public void SelectDiet()
+        {
+            IoCManager.Resolve<IRobustRandom>();
+            //Picks a random diet, if can pick one
+            if (DietPick is not null)
+            { 
+               SelectedDiet = _random.Pick(DietPick);
+            }
+        }
+
+        //A placeholder function for advancing specimen growth
+        //Should only be used to increase size in the adult stage outside of the tube
+        public void SpecimenGrow() 
+        {
+            GrowthState++;
+        }
```

Comment:
This is a crime.

Reply:
Is it just painful to look at or does this impact performance or cause bugs??